client/consensus/babe/src/authorship.rs

```diff
@@ -124,7 +124,7 @@ pub(super) fn secondary_slot_author(
 fn claim_secondary_slot(
 	slot_number: SlotNumber,
 	epoch: &Epoch,
-	keystore: &KeyStorePtr,
+	key_pairs: &Vec<(AuthorityPair, usize)>,
```

Comment:
`&Vec` in Rust is usually a bad idea. Use a slice instead.

Reply:
based on the usage here you could even use `impl IntoIterator<Item=(...)>` instead of the explicit `&Vec`.